\documentclass[12pt]{article}
\usepackage[margin=1in]{geometry}
\usepackage[utf8]{inputenc}
\usepackage{setspace}
\usepackage{amsmath,amsthm,amsfonts,mathtools,color,amssymb}
\usepackage{tikz}
\usepackage{mathrsfs}
\usepackage{biblatex}

\title{Masters Project Report}
\author{Emily Palmer}
\date{Spring 2021}

\addbibresource{bib.bib}
\doublespacing

\begin{document}

\maketitle

\begin{abstract}
  Application of a recent method for microbiome association studies to larger datasets provided computational challenges. Several aspects of the method to more general data were developed for use beyond the case study provided in the initial paper.
\end{abstract}


\tableofcontents



\section{Introduction}

\subsection{About microbiome data}
The human gut contains millions of microbes.
Genomic research on microbiome sequencing data has provided numerous insights to ...*** Studies have associated the human gut microbiome to irritable bowel syndrome, inflammatory bowel disease, cancers, diabetes, and obesity, with more studies coming out with more interesting results. \cite{kinross2008human} Bidirectional interactions have been found between the gut and the central nervous system, colloquially referred to as the "gut-brain-axis". \cite{mayer2015gut}

These studies show the microbiome to be a research rich and important area of research for advances in medical, and our understanding of ourselves. (FIX)


One technique for microbiome analysis is sequencing the 16S ribosomal rRNA gene. This gene is a useful marker for



Microbiome data commonly is sampled from fecal samples, but oral, skin, vaginal, etc are also common areas (CITE)


Microbiome data commonly arises from 16S ribosomal ribonucleic acid RNA (rRna) sequencing studies. Microbiome sequencing data commonly consists of the count of reads on operational taxonomic units (OTUs) for each sample. Operational taxonomic units are.... Methods for microbiome analysis can either be focused on a single OTU or multiple OTUs simultaneously.



\subsubsection{Issues with microbiome data}

 OTU data present many obstacles to analysis using standard statistical methods.

 There are several aspects common to microbiome data that violate the assumptions of common statistical tests and regression models.

Frequently, the sampling depth (FINISH) is not consistent across samples. One technique to adjust for this is rarifying (CITE, FINISH)

First, the count for a given OTU does not necessarily represent the overall count. In other words, these counts represent relative abundance, not absolute abundance. OTU data is thus compositional in nature. Often, OTU counts are converted into a proportion to represent the relative abundance of that OTU in the sample.

Secondly OTU data is very sparse. The library size of different samples can vary a lot. For many OTUs, the total reads is zero. Thus microbiome data contains excessive zeros, which much be accounted for in the analysis.
Adjustments can be made using zero inflated models, or two-part models that separately model the presence of an OTU, and then next the abundance of present OTUs.




\subsubsection{Organizing OTUs onto a tree}

(FIND CITATIONS)

All life can be linked back to a biologic tree that describes how each organism is related to each other.

Two types of trees that can arise in microbial studies are phylogenetic trees and taxonomic trees.

Using knowledge of a given OTU (FIX), it can provide insight into correlations that exist between different read counts present in samples, and this correlation can be helpful in modeling overall association with predictors of interest.




\subsection{Generalized Estimating Equations}

Generalized Estimating Equations (GEEs) are a useful tool for correlated data, such as those that arise from longitudinal data. GEEs were introduced to extend generalized linear models to longitudinal data. \cite{liang1986longitudinal} Consider a response variable $y_{it}$, and a $p \times 1$ vector of covariates $x_{it}$, where $i = 1, \ldots , K$ represents the index of clusters, and $t = 1, \ldots , n_i$ represent the number of time points, or more generally, different values, or repeated measures in an individual cluster.

GEEs work without specifying the joint distribution of observations, similarly to quasi-likelihood approaches.







\section{About the method ()}

The Microbiome Taxonomic Longitudinal Correlation (MLTC) model was introduced in 2020 by Chen and Xu\cite{chen2020generalized} as a way to estimate correlations between OTUs and associations between a predictor an, that can be used on longitudinal and repeated measure data. Coefficients are estimated using a two part GEE model that uses a correlation structure based on OTU correlations and longitudinal and repeated measure correlations.

\subsection{Accounting for Correlation}


\subsubsection{Model Data structure}
Commonly, OTU data is stored in two tables. Consider a dataset with $k$ individuals. Each individual $i$ has $n_i$ different samples taken from it. In total there are $M = \sum_{i=1}^k n_j$ samples taken.
Additionally, each sample has $N$ OTUs measured, and $p$ predictors. The first table contains counts for OTU values, with OTU labels as rows and samples as columns. This table will have overall dimension $N \times M$ The second table contains metadata (predictor) information, with samples as rows and predictor values as columns, having dimension $M \times p$.

In this model, the response is the OTU value (either presence absence, or transformed relative abundance). The reponse $\mathbf{y} = (\mathbf{y}_1, \ldots, \mathbf{y}_k)^t$




\subsubsection{Taxa correlation}
\subsubsection{Longitudinal and repeated measure correlation}
\subsubsection{Interative correlation matrix }
\subsubsection{Two-part GEE model}


\section{Extensions to the method}
\subsection{"Missing data"}
\subsection{Generating the taxa correlation from microbiome data}
\subsection{How to deal with unlabeled taxa}
\subsection{Application to American gut data}
\subsubsection{About the American gut dataset}
\subsubsection{Filtering the American gut dataset}
\subsubsection{Computational challenges}


\section{Conclusion}
\subsection{Future work}







\printbibliography


\end{document}
