\documentclass[12pt]{article}
\usepackage[margin=1in]{geometry}
\usepackage[utf8]{inputenc}

\usepackage{amsmath,amsthm,amsfonts,mathtools,color,amssymb}
\usepackage{tikz}
\usepackage{mathrsfs}
\usepackage{biblatex}

\title{Masters Project Report}
\author{Emily Palmer}
\date{Spring 2021}

\addbibresource{bib.bib}

\begin{document}

\maketitle

\begin{abstract}
  Application of a recent method for microbiome association studies to larger datasets provided computational challenges. Several aspects of the method to more general data were developed for use beyond the case study provided in the initial paper.
\end{abstract}


\tableofcontents



\section{Introduction}

\subsection{About microbiome data}

One technique for microbiome analysis is sequencing the 16S ribosomal rRNA gene. This gene is a useful marker for




Microbiome data generated from 16S ribosomal RNA (rRna) sequencing present many obsticals to analysis using standard statistical methods. There are several aspects common to microbiome data that violate the assumptions of common statistical tests.

\subsubsection{Issues with microbiome data}
\begin{enumerate}
  \item sparcity
  \item
\end{enumerate}


\subsection{Generalized Estimating Equations}
\cite{chen2020generalized}

\section{About the GEE method ()}



\section{Extensions to the method}
\subsection{"Missing data"}
\subsection{Generating the taxa correlation from microbiome data}
\subsection{How to deal with unlabeled taxa}
\subsection{Application to American gut data}
\subsubsection{About the American gut dataset}
\subsubsection{Filtering the American gut dataset}
\subsubsection{Computational challenges}


\section{Conclusion}








\printbibliography


\end{document}
